fal/views/teams.py

```diff
@@ -0,0 +1,217 @@
+            f.write(HEADCOUNT_CONC_TEXT.format(len(teams)))
+
+
+def team_overview(season_str: str = season_str, year: int = year) -> None:
+    """
+    Creates a formatted forum post for the team overview thread.
+    """
+    week: int = config.getint('weekly info', 'current-week')
+    with session_scope() as session:
+        teams: List[Team] = get_team_from_season(season_str, year, session)
+        with open("lists/team_overview.txt", "w", encoding="utf-8") as f:
+            f.write(f"Team List - FAL {season_str.capitalize()} {year}\n\n\n")
+            for team in sorted(teams, key=lambda t: t.name):
+                base_query = session.query(TeamWeeklyAnime, Anime). \
+                    filter(TeamWeeklyAnime.team_id == team.id). \
+                    filter(TeamWeeklyAnime.anime_id == Anime.id). \
+                    filter(TeamWeeklyAnime.week == week)
+                active_anime = base_query.filter(
+                    TeamWeeklyAnime.bench.is_(False)).all()
+                bench_anime = base_query.filter(
+                    TeamWeeklyAnime.bench.is_(True)).all()
+                f.write(f"{team.name}\n---------------------------------\n")
+                # list all active series
+                for anime in sorted(active_anime, key=lambda a: a[1].name.lower()):
+                    f.write(f"{anime[1].name}\n")
+                f.write("\n")
+                # list all bench series
+                for anime in sorted(bench_anime, key=lambda a: a[1].name.lower()):
+                    f.write(f"{anime[1].name}\n")
+                f.write("\n\n")
+            f.write("[/spoiler]")
+
+
+def team_stats(season_str: str = season_str, year: int = year, prep: bool = True) -> None:
+    """
+    Creates a statistic of the titles distribution for the team overview thread.
+    This function can also be used during the game to obtain the distribution
+    of the current week.
+    @param prep during the real game (False) or before the game (True)
+    """
+    week: int = config.getint('weekly info', 'current-week')
+    if not prep:
+        filename = f"lists/team_stats_{week}.txt"
+        raise NotImplementedError("Haven't implemented stats during real game")
+    else:
+        filename = "lists/team_stats.txt"
+    with session_scope() as session:
+        teams: List[Team] = get_team_from_season(season_str, year, session)
+        season: Season = get_season_from_database(season_str, year, session)
+        query = session.query(Anime).filter(Anime.season_id == season.id)
+        anime_list: List[Anime] = query.all()
+        stats: Dict[int, AnimeTeamCount] = {
+            a.id: AnimeTeamCount(0, 0) for a in anime_list}
+        for team in teams:
+            base_query = session.query(TeamWeeklyAnime). \
+                filter(TeamWeeklyAnime.team_id == team.id). \
+                filter(TeamWeeklyAnime.week == week)
+            series: List[TeamWeeklyAnime] = base_query.all()
+            for anime in series:
```

Comment:
Then you can chain the results into `collections.Counter(dict(list_of_pairs))`

Reply:
I'd imagine that pattern would come in handy for other analytics views we may implement later

Reply:
I haven't written any tests yet, so I'll take a look at it.

Reply:
you'd probably have to count all TeamWeeklyAnime filtered by the week, grouped by anime. then run the query again with the bench filter

Reply:
Wait are you saying I should count the number of times the list of 7 TeamWeeklyAnime shows up in the database for that week? Since we want to see the number of same teams.

Reply:
We also need the names of the users with the same teams so I am not sure how it work.

Reply:
it looks like the output of this function is how many teams picked that anime for this week, and how many of the teams put it in active, right?

Reply:
Ah you're right, I thought we were looking at team_dist, my bad.

Reply:
maybe this function could be renamed to something more distinct? like maybe `anime_distribution`?

Reply:
It's about the same teams though, so I think team_dist is a better name.

Reply:
Lol but once again this isn't `team_dist`

Reply:
I'm sorry I'm really out of it today, I saw anime_distribution and I thought you meant this function as in team_dist for some reason.